casper/src/main/scala/coop/rchain/casper/util/rholang/SystemDeploy.scala

```diff
@@ -19,7 +19,7 @@ abstract class SystemDeploy(initialRand: Blake2b512Random) {
   type Result
   type Env
 
-  final val rand = initialRand.copy()
+  val rand = initialRand.copy()
```

Comment:
Again, I'm unsure why you would want to use a different RNG than the one passed to `SystemDeploy`.

Reply:
This was related to what I described here: https://github.com/rchain/rchain/pull/2781#discussion_r365625061